syntax = "proto3";

package NKikimrBlobStorage;

// Enum defining PDisk underlying drive type (rotational or solid state)
enum EPDiskType {
    ROT = 0; // rotational drives (HDD)
    SSD = 1; // solid state drives (SSD)
    NVME = 2; // PCIe-connected solid state drives (NVMe SSD)
    UNKNOWN_TYPE = 3; // used if device type is unknown or if group consists of different PDisk device types
}

// Virtual group life cycle state
enum EVirtualGroupState {
    NEW = 0; // just created
    CREATE_FAILED = 1; // error while creating VirtualGroup
    WORKING = 2; // operational
    DELETING = 3; // going to delete from Hive
}

message TGroupDecommitStatus {
    enum E {
        NONE = 0; // no decomission
        PENDING = 1; // decommission machinery is starting
        IN_PROGRESS = 2; // decomission underway
        DONE = 3; // group decomission complete
    }
}
